include/ppx/geometry.h

```diff
@@ -182,9 +176,13 @@ class Geometry
 
             // Current size
             size_t offset = mData.size();
-            // Allocate storage for incoming data
+            // Reserve double storage if necessary
+            if ((offset + sizeOfValues) > mData.capacity()) {
+                mData.reserve(std::max<size_t>(offset + sizeOfValues, mData.size() * 2));
```

Comment:
Why using reserve?
On linux, debug build:
 - before this PR: 30s
 - with this PR: 11s
 - with this PR, keeping the resize as in #312 : 7s

Reply:
I thought I had a reason for this but I can't remember it anymore, so I think it's fine to use resize if the functionality is the same.

I am calling the `Buffer::SetSize()` that is removed in #312 though, so I'd prefer if that function is kept in.

Reply:
Sure, I removed `SetSize()` because we had no users, but since you are now using it, I'll add it back.